### PR TITLE
Slack Guidelines

### PR DESCRIPTION
**Pull request by kvnkho**
## Summary

Guidelines for Slack users. This replaces the old Community page as a lot of the content there was stale.

### CODE REVIEW

File: docs/core/community.md [hunk 1]
```diff
@@ -8,31 +8,51 @@ We welcome issues, contributions and discussion from all users, regardless of ba
 
 Please consider [Chris White](https://github.com/cicdw) the main point of contact for the Prefect repo.
 
-## Discussion
+# Slack
 
-- We have a public [Slack](https://www.prefect.io/slack) for chatting about Prefect, asking questions, and sharing tips.
-- For feature requests or bug reports, please open a [new GitHub issue](https://github.com/PrefectHQ/prefect/issues/new).
-- Important architectural and design decisions are memorialized in [Prefect Improvement Notices (PINs)](/core/PINs/PIN-01-Introduce-PINs.html).
+We have a public [Slack](https://prefect-slackin.herokuapp.com/) for chatting about Prefect, asking questions, and sharing tips.
```
Comment (madkinsz) [inline docs/core/community.md:13]: Is this the heroku app that we just replaced? cc @znicholasbrown

Comment (znicholasbrown) [reply docs/core/community.md:13]: Yup it is - use this instead:
```suggestion
We have a public [Slack](https://prefect.io/slack) for chatting about Prefect, asking questions, and sharing tips.
```

File: docs/core/community.md [hunk 2]
```diff
@@ -8,31 +8,53 @@ We welcome issues, contributions and discussion from all users, regardless of ba
...
+**Rule 4: Write high quality questions**
+
+The Prefect community is here to support you. That said, it is significantly easier to answer well-researched and clearly-written questions. Even adding potentially relevant links to a post helps tremendously.
+Informative Slack threads are archived by our resident bot Marvin. Having well-written threads helps future users encountering the same problem.
+
+Oftentimes your question may have been answered somewhere else; some good resources to start looking before asking a question:
+* Previous Slack messages (we do our best to archive relevant threads to GitHub!)
+* [Prefect documentation](https://docs.prefect.io/core/development/documentation.html)
+* [Github issues](https://github.com/PrefectHQ/prefect/issues)
+* StackOverflow
```
Comment (cicdw) [inline docs/core/community.md:44]:
```suggestion
* [StackOverflow](https://stackoverflow.com/questions/tagged/prefect)
```

**Review by cicdw — CHANGES_REQUESTED** [review]
Some suggestions for linking to other resources, otherwise LGTM!

**Review by cicdw — APPROVED** [review]

**Pull request merged**